packages/core/src/routes/role.test.ts

```diff
@@ -19,4 +30,38 @@ describe('role routes', () => {
     expect(response.status).toEqual(200);
     expect(response.body).toEqual([mockRole]);
   });
+
+  it('POST /roles', async () => {
+    const { name, description } = mockRole;
+
+    const response = await roleRequester.post('/roles').send({ name, description });
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(mockRole);
+    expect(hasRoleWithName).toHaveBeenCalled();
+  });
+
+  it('GET /roles/:id', async () => {
+    findRoleById.mockResolvedValueOnce(mockRole);
+    const response = await roleRequester.get(`/roles/${mockRole.id}`);
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(mockRole);
+  });
+
+  it('PATCH /roles/:id', async () => {
+    findRoleById.mockResolvedValueOnce(mockRole);
+    const response = await roleRequester
+      .patch(`/roles/${mockRole.id}`)
+      .send({ description: 'new' });
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual({
+      ...mockRole,
+      description: 'new',
+    });
+  });
```

Comment:
maybe should add the case that the role name you want to change has already been occupied.

Reply:
added